presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -471,15 +471,6 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
         String writerImplementation = hiveFileWriter.getClass().getName();
 
         Consumer<HiveWriter> onCommit = hiveWriter -> {
-            Optional<Long> size;
```

Comment:
Let's preserve the existing behavior of not failing the query by changing this to
```java
Optional<Long> size;
try {
    size = hiveWriter.getWrittenBytes();
}
catch (RuntimeException e) {
    // Do not fail the query if file system is not available
    size = Optional.empty();
}
```
Then we don't need to worry about retry inside of `RecordFileWriter`. We can remove this (and the stat call) once we implement position fetching for all writers.

Reply:
Yes, makes sense to keep the existing behavior intact for now.